src/main/scala/org/apache/mesos/chronos/scheduler/jobs/JobStats.scala

```diff
@@ -124,106 +166,157 @@ class JobStats @Inject()(clusterBuilder: Option[Cluster.Builder], config: Cassan
-                job.name,
-                job.owner,
-                job.schedule,
-                taskStatus.getState.toString,
-                taskStatus.getSlaveId.getValue,
-                attempt: java.lang.Integer
-              ))
-            case job: DependencyBasedJob =>
-              val query =
-                s"INSERT INTO ${config.cassandraTable()} (id, ts, job_name, job_owner, job_parents, task_state, slave_id, attempt) VALUES (?, ?, ?, ?, ?, ?, ?, ?) USING TTL ${config.cassandraTtl()}"
-              val prepared = statements.getOrElseUpdate(query, {
-                session.prepare(
-                  new SimpleStatement(query).setConsistencyLevel(ConsistencyLevel.valueOf(config.cassandraConsistency())).asInstanceOf[RegularStatement]
-                )
-              })
-              val parentSet: java.util.Set[String] = job.parents.asJava
-              session.execute(prepared.bind(
-                taskStatus.getTaskId.getValue,
-                new java.util.Date(),
-                job.name,
-                job.owner,
-                parentSet,
-                taskStatus.getState.toString,
-                taskStatus.getSlaveId.getValue,
-                attempt: java.lang.Integer
-              ))
-          }
-        case None =>
-      }
-    } catch {
-      case e: NoHostAvailableException =>
-        resetSession()
-        log.log(Level.WARNING, "No hosts were available, will retry next time.", e)
-      case e: QueryExecutionException =>
-        log.log(Level.WARNING, "Query execution failed:", e)
-      case e: QueryValidationException =>
-        log.log(Level.WARNING, "Query validation failed:", e)
+    updateJobState(job.name, CurrentState.idle)
+    job match {
+      case job: ScheduleBasedJob =>
+        insertToStatTable(
+            id=Some(taskStatus.getTaskId.getValue),
+            timestamp=Some(new java.util.Date()),
+            jobName=Some(job.name),
+            jobOwner=Some(job.owner),
+            jobSchedule=Some(job.schedule),
+            taskState=Some(taskStatus.getState.toString),
+            slaveId=Some(taskStatus.getSlaveId.getValue),
+            attempt=Some(attempt),
+            jobParents=None, message=None,
+            isFailure=None)
+      case job: DependencyBasedJob =>
+        val parentSet: java.util.Set[String] = job.parents.asJava
+        insertToStatTable(
+            id=Some(taskStatus.getTaskId.getValue),
+            timestamp=Some(new java.util.Date()),
+            jobName=Some(job.name),
+            jobOwner=Some(job.owner),
+            jobParents=Some(parentSet),
+            taskState=Some(taskStatus.getState.toString),
+            slaveId=Some(taskStatus.getSlaveId.getValue),
+            attempt=Some(attempt),
+            jobSchedule=None, message=None,
+            isFailure=None)
     }
   }
 
   def jobFailed(job: BaseJob, taskStatus: TaskStatus, attempt: Int) {
-    try {
+    updateJobState(job.name, CurrentState.idle)
```

Comment:
A lot of this code is duplicated between the cases. Let's separate out the common logic.